### PR TITLE
ARM64: Insure that the upper bits of the stack slots are zero-ed

### PR DESCRIPTION
**Pull request by briansull**

### CODE REVIEW

**Comment by briansull** [comment]
@dotnet-bot Test  Windows_NT arm64 Cross Checked Innerloop Build and Test

File: src/jit/codegenarmarch.cpp [hunk 1]
```diff
@@ -597,14 +597,35 @@ void CodeGen::genPutArgStk(GenTreePutArgStk* treeNode)
...
+                    break;
+                case 4:
+                    // zero out the upper 4 bytes of the stack slot
+                    emit->emitIns_S_R(INS_str, EA_4BYTE, REG_ZR, varNumOut, argOffsetOut + 4);
+                    break;
+                default:
+                    noway_assert(!"Invalid size for scalar arg");
+            }
+        }
+#endif // !_TARGET_ARM64
```
Comment (adityamandaleeka) [inline src/jit/codegenarmarch.cpp]: `// _TARGET_ARM64`

**Comment by briansull** [comment]
@jashook  @CarolEidt PTAL

**Comment by AndyAyersMS** [comment]
Any chance you can add a test case?

**Comment by briansull** [comment]
It is manged C++ program.

**Comment by briansull** [comment]
@dotnet-bot retest this

**Comment by jashook** [comment]
Can it be checked in as an ilproj?

**Comment by briansull** [comment]
I don't think so, manged C++ compiles to a single architecture.

**Comment by briansull** [comment]
Test Case:

The C++/CLI code below fails on ARM64 because we end up passing a garbage parameter in x1 to strcat.
...
#include <stdio.h>
#include <string.h>
#include <stdarg.h>

void CallVar(char *buf, const char *str, ...)
{
    va_list args;

    va_start(args, str);

    const char *s = str;
    while (s) {
        strcat(buf, s);
        s = va_arg(args, const char*);
    }
}

int __cdecl main(int, const char**)
{
    char buf[1024];

    buf[0] = '\0';

    // VSO403979: Under C++ compiles, the NULL gets translated to the literal
    // '0', which is an i32. The compiler just writes the low 4 bytes into the
    // stack slot, whereas we'd prefer that the full 8 bytes are written so that
    // it is correctly interpretable as a null pointer.
    CallVar(buf, "0", "1", "2", "3", "4", "5", "6", "7", NULL);

    if (!strcmp(buf, "01234567")) {
        printf("pass\n");
    }

    return 0;
}
```

**Pull request closed without merging**